### PR TITLE
Dycore paper updates in how we compute SHF and LHF

### PR DESCRIPTION
**Pull request by trontrytel**
Updates following section 3.6.4 in the dycore paper.

Tests work for me locally when using the Thermodynamics v0.13 branch. This PR should be merged after we do the Thermodynamics release.

cc: @kmdeck - will it affect the land models?

### CODE REVIEW

**Comment by trontrytel** [comment]
Sorry, none of the tests will work until we have Thermodynamics v0.13, because I put it as a requirement.

**Comment by kmdeck** [comment]
@trontrytel In simulations where land is forced by a prescribed atmosphere, land needs to compute its own fluxes. In principle that means we just call SF functions, but instead right now we do that and then adjust the SF fluxes as follows:
https://github.com/CliMA/ClimaLand.jl/blob/a746febe976c4dcab555c2afb8dbbe0483e169b4/src/shared_utilities/drivers.jl#L367-L372

This is actually more similar to the SF computation in this PR, so that is good. We had switched to this form because it was easier to debug/understand than the previous SF computation (e.g. SHF doesnt depend on E). But note that we also need to account for surface resistance in the fluxes, and SF does not handle that right now, so we do it "offline". 

To couple the full land model to atmos, we need to adjust SF to account for resistance.  We should also adjust our standalone land model to use the output of SF directly. This will make coupled runs easier to debug.

File: src/SurfaceFluxes.jl [hunk 1]
```diff
@@ -683,10 +683,8 @@ function sensible_heat_flux(param_set, Ch, sc::Union{ValuesOnly, Coefficients},
     T_in = TD.air_temperature(thermo_params, ts_in(sc))
     T_sfc = TD.air_temperature(thermo_params, ts_sfc(sc))
     ΔT = T_in - T_sfc
-    hd_sfc = cp_d * (T_sfc - T_0) + R_d * T_0
     ΔΦ = grav * Δz(sc)
-    E = evaporation(param_set, sc, Ch)
-    return -ρ_sfc * Ch * windspeed(sc) * (cp_m * ΔT + ΔΦ) - (hd_sfc) * E
+    return -ρ_sfc * Ch * windspeed(sc) * (cp_m * ΔT + ΔΦ)
```
Comment (szy21) [inline src/SurfaceFluxes.jl:687]: I double-checked with Tapio about this. Here we assume that cp_m at the surface and the interior are the same (which is an ok assumption). To be more accurate we should use ΔDSE here as in the dycore paper (you can get cp_m at the surface from the surface thermo state). One caveat is that I don't know whether it is correct for land, we can leave a comment about land for now.

Comment (trontrytel) [reply src/SurfaceFluxes.jl]: Changed!

**Review by szy21 — APPROVED** [review]

**Review by akshaysridhar — APPROVED** [review]

**Pull request merged**